catalog/nsili/catalog-nsili-endpoint/src/test/java/com/connexta/alliance/nsili/endpoint/TestGetRelatedFilesRequestImpl.java

```diff
@@ -0,0 +1,553 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+import org.apache.http.Header;
+import org.apache.http.HeaderIterator;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.ProtocolVersion;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.params.HttpParams;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.omg.CORBA.NO_IMPLEMENT;
+
+import com.connexta.alliance.nsili.common.UCO.FileLocation;
+import com.connexta.alliance.nsili.common.UCO.NameListHolder;
+import com.connexta.alliance.nsili.common.UCO.State;
+import com.connexta.alliance.nsili.endpoint.requests.GetRelatedFilesRequestImpl;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+
+public class TestGetRelatedFilesRequestImpl {
+
+    private static final int TEST_PORT = 21000;
+
+    private GetRelatedFilesRequestImpl relatedFilesRequest;
+
+    private HttpClient mockHttpClient = mock(HttpClient.class);
+
+    @Before
+    public void setUp() throws Exception {
+        List<Metacard> testMetacards = getTestMetacards();
+        FileLocation location = getTestFileLocation();
+        relatedFilesRequest = new GetRelatedFilesRequestImpl(testMetacards,
+                location,
+                "THUMBNAIL",
+                TEST_PORT);
+        relatedFilesRequest.setHttpClient(mockHttpClient);
+        setupMocks();
+    }
+
+    @Test
+    public void testComplete200() throws Exception {
+        when(mockHttpClient.execute(any(HttpUriRequest.class))).thenReturn(getGoodResponse(200));
+        NameListHolder locations = new NameListHolder();
+        State state = relatedFilesRequest.complete(locations);
+        assertThat(state, is(State.COMPLETED));
+        assertThat(locations, notNullValue());
+        assertThat(locations.value, notNullValue());
+        assertThat(locations.value.length, is(1));
+    }
+
+    @Test
+    public void testComplete201() throws Exception {
+        when(mockHttpClient.execute(any(HttpUriRequest.class))).thenReturn(getGoodResponse(201));
+        NameListHolder locations = new NameListHolder();
+        State state = relatedFilesRequest.complete(locations);
+        assertThat(state, is(State.COMPLETED));
+        assertThat(locations, notNullValue());
+        assertThat(locations.value, notNullValue());
+        assertThat(locations.value.length, is(1));
+    }
+
+    @Test
+    public void testComplete202() throws Exception {
+        when(mockHttpClient.execute(any(HttpUriRequest.class))).thenReturn(getGoodResponse(202));
+        NameListHolder locations = new NameListHolder();
+        State state = relatedFilesRequest.complete(locations);
+        assertThat(state, is(State.COMPLETED));
```

Comment:
These 4 lines seen to be repeated a lot, can we consolidate this into an `assertState` method?